param/__init__.py

```diff
@@ -1007,8 +1008,15 @@ def _validate_value(self, val, allow_None):
                                  "Boolean value or None, not %s."
                                  % (self.name, val))
         elif not isinstance(val, bool):
-            raise ValueError("Boolean parameter %r must be True or False, "
-                             "not %s." % (self.name, val))
+            if self.name is not None:
+                raise ValueError("Boolean parameter %r must be True or False, "
+                                 "not %s." % (self.name, val))
+            else:
+                raise ValueError("Boolean parameter must be True or False, "
+                                 "not %s." % val)
+
```

Comment:
```suggestion
            name = " " + self.name if self.name else ""
            raise ValueError("Boolean parameter%r must be True or False,  not %s." % (name, val))
```

Reply:
👍 Nice. There was an extra space in the suggestion so I did in locally, had to deal with %r on `" " + self.name` being slightly off as well.